tests/tests/operation_tests.cpp

```diff
@@ -6771,9 +6771,13 @@ BOOST_FIXTURE_TEST_SUITE(operation_tests, clean_database_fixture)
             delegation = db->find<vesting_delegation_object, by_delegation>(std::make_tuple(op.delegator, op.delegatee));
             BOOST_CHECK(delegation == nullptr);
 
+            BOOST_CHECK_EQUAL(db->get_account("bob").received_vesting_shares, ASSET_GESTS(0));
+
+            BOOST_CHECK(db->get_account("alice").delegated_vesting_shares != ASSET_GESTS(0));
```

Comment:
1. Please, use `BOOST_CHECK_NE` here, it will log actual values of args if fail.
2. Here we know if delegated_vests do not set to 0 instantly after operation, but there is no test to check that setting to 0 happens at the time we expect. Code can break so zeroing happens in the middle of `STEEMIT_CASHOUT_WINDOW_SECONDS`, and this test will say nothing about bug. Please, add check of delegated_vests at block just before expiration.

Reply:
@zxcat 
1. OK.
2. How can it happen in the middle? This way we will write our tests infinitely, no?

Reply:
> How can it happen in the middle?

bugs are always unexpected

> This way we will write our tests infinitely, no?

Test should cover code it tests, and there is not so much logic to cover.


> And why check it in reject expiration tests? It was used in old expiration code, too.

Because it used in reject. If you add separate function, used by both normal expiration and reject, then only one test is enough. It can be either in old expiration tests or in reject tests. But if you copy and paste code, there should be separate test. Because one copy-pasted block can be changed (by mistake), and proper test should detect bug.

Actually, such good coverage is not high priority for now. You can skip this suggesition if there are problems with it